Code/Framework/AzCore/AzCore/Jobs/Internal/JobManagerWorkStealing.cpp

```diff
@@ -77,7 +78,135 @@ Job* WorkQueue::TryStealFront()
 
     return nullptr;
 }
+#else // JOBMANAGER_LEGACY_WORKQUEUE
+struct StatusPayload
+{
+    StatusPayload(uint64_t status)
+    {
+        memcpy(this, &status, sizeof(status));
+    }
+
+    StatusPayload& operator=(uint64_t status)
+    {
+        memcpy(this, &status, sizeof(status));
+        return *this;
+    }
```

Comment:
We should comment that this works due to O3DE only running on little endina systems.

Reply:
Hey actually I don't think system endianness will affect this code since we memcpy in and out when modifying bits via this payload class.